source/extensions/filters/http/cache/cache_headers_utils.cc

```diff
@@ -0,0 +1,182 @@
+#include "extensions/filters/http/cache/cache_headers_utils.h"
+
+#include <array>
+#include <string>
+
+#include "envoy/common/time.h"
+
+#include "absl/algorithm/container.h"
+#include "absl/strings/ascii.h"
+#include "absl/strings/numbers.h"
+#include "absl/strings/str_split.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+
+// Utility functions used in RequestCacheControl & ResponseCacheControl
+namespace {
+OptionalDuration parseDuration(absl::string_view s) {
+  OptionalDuration duration;
+  // Strip quotation marks if any
+  if (!s.empty() && s.front() == '"' && s.back() == '"') {
```

Comment:
What happens if s is (somehow) a string_view of length 1 containing the single character '"'? Seems like this might then call substr with a negative length.

Reply:
abseil handles this internally, if `s = "` , s.substr(1, -1) is an empty string

Reply:
However, I am not sure if that's a defined behavior that we can count on. Probably worth a check.
Thanks!